src/group_theory/nilpotent.lean

```diff
@@ -577,6 +577,46 @@ lemma nilpotency_class_quotient_le (H : subgroup G) [H.normal] [h : is_nilpotent
 lemma derived_le_lower_central (n : ℕ) : derived_series G n ≤ lower_central_series G n :=
 by { induction n with i ih, { simp }, { apply general_commutator_mono ih, simp } }
 
+lemma _root_.comm_group.center_eq_top {G : Type*} [comm_group G] : center G = ⊤ := begin
+  ext x,
+  suffices : x ∈ center G, by simpa,
+  intro y,
+  group,
+end
+
+lemma _root_.group.comm_group_of_center_eq_top (h : center G = ⊤) : comm_group G :=
+begin
+  have h' : ∀ x, x ∈ center G, by simp [h],
+  exact { mul_comm := λ x y, h' y x, .. (_ : group G) }
+end
```

Comment:
I will move them to `subgroup` next to the other lemmas about `center G` once they are fine (easier to edit in the leaf module until then)